PubMatic/src/main/java/com/applovin/mediation/adapters/PubMaticMediationAdapter.java

```diff
@@ -509,5 +543,155 @@ public void onAdClicked(@NonNull final POBBannerView view)
+            if(mainImageAsset != null)
+            {
+                final String mainImageUrl = mainImageAsset.getImageURL();
+                builder.setMainImage(new MaxNativeAd.MaxNativeAdImage(Uri.parse(mainImageUrl)));
+            }
+
+            // Map options view.
+            builder.setOptionsView(pobNativeAd.getAdInfoIcon());
+
+            // Create NativeAd and delegate onNativeAdLoaded callback.
+            MAXPubMaticNativeAd maxNativeAd = new MAXPubMaticNativeAd(builder, pobNativeAd, maxNativeAdAdapterListener);
+            maxNativeAdAdapterListener.onNativeAdLoaded(maxNativeAd, null);
+        }
+
+        @Override
+        public void onFailedToLoad(@NonNull POBNativeAdLoader pobNativeAdLoader, @NonNull POBError pobError)
+        {
+            MaxAdapterError maxAdapterError = toMaxError(pobError);
+
+            log( "Native ad failed to load with error (" + maxAdapterError + ")" );
+            maxNativeAdAdapterListener.onNativeAdLoadFailed(maxAdapterError);
+        }
+    }
+
+
+
+    private class MAXPubMaticNativeAd extends MaxNativeAd
+    {
+        @NonNull
+        private final POBNativeAd mNativeAd;
+        @NonNull
+        private final MaxNativeAdAdapterListener maxNativeAdAdapterListener;
+
+        public MAXPubMaticNativeAd(@NonNull Builder builder, @NonNull POBNativeAd nativeAd, @NonNull MaxNativeAdAdapterListener maxNativeAdAdapterListener)
+        {
+            super( builder );
+            this.mNativeAd = nativeAd;
+            this.maxNativeAdAdapterListener = maxNativeAdAdapterListener;
+        }
+
+        @Override
+        public boolean prepareForInteraction(List<View> list, ViewGroup viewGroup)
+        {
+            mNativeAd.registerViewForInteraction( viewGroup, list, new POBNativeAdListener()
+            {
+                @Override
+                public void onNativeAdRendered(@NonNull POBNativeAd pobNativeAd) {
+                    // No action required.
+                }
+
+                @Override
+                public void onNativeAdRenderingFailed(@NonNull POBNativeAd pobNativeAd, @NonNull POBError error) {
+                    log( "Native ad failed to show with error (" + error + ")" );
+                }
+
+                @Override
+                public void onNativeAdImpression(@NonNull POBNativeAd pobNativeAd) {
+                    log( "Native ad impression" );
+                    maxNativeAdAdapterListener.onNativeAdDisplayed(null);
+                }
+
+                @Override
+                public void onNativeAdClicked(@NonNull POBNativeAd pobNativeAd) {
+                    log( "Native ad clicked" );
+                    maxNativeAdAdapterListener.onNativeAdClicked();
+                }
+
+                @Override
+                public void onNativeAdClicked(@NonNull POBNativeAd pobNativeAd, @NonNull String s) {
+                    // No action required.
```

Comment:
We can call `maxNativeAdAdapterListener.onNativeAdClicked();` on this callback as well.